buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyGen.groovy

```diff
@@ -0,0 +1,422 @@
+                            String packageName = String.join(PACKAGE_SEPARATOR, ASSEMBLY_PACKAGE_BASE, kamName)
+
+                            DumpSettings dumpSettings = DumpSettings.builder()
+                                .setIndent(4)
+                            // FLOW (default)serializes any embedded maps.
+                            // This will recurse.
+                                .setDefaultFlowStyle(FlowStyle.BLOCK)
+                                .build()
+                            Dump dumper = new Dump(dumpSettings)
+
+                            // Now, extract the route template.  This will become our route document after we change
+                            // kamelet:source or kamelet:sink to vantiq server URLs
+
+                            //noinspection unchecked
+                            Map<String, Object> template = (Map<String, Object>) ((Map<String, Object>) yamlMap
+                                .get('spec')).get('template')
+                            String originalTemplateString = dumper.dumpToString(template)
+                            log.info('\n>>>   Original Template as Yaml:\n{}', originalTemplateString)
+
+                            template = constructRouteText(template)
+
+                            // Now, generate & populate results
+                            Path thisKameletDirectory = Paths.get(generateBase.toAbsolutePath().toString(), kamName)
+                            if (!Files.exists(thisKameletDirectory)) {
+                                log.info('Creating kamelet-specific project directory: {}', thisKameletDirectory)
+                                thisKameletDirectory = Files.createDirectories(thisKameletDirectory)
+                            }
+                            log.info('\nResults for Kamelet: {}', kamName)
+                            log.info('\n>>>   Properties:')
+                            props.forEach((name, val) -> {
+                                //noinspection unchecked
+                                log.info('    Name: {} :: [title: {}, type: {}, desc: {}]', name,
+                                    ((Map<String, String>) val).get('title'),
+                                    ((Map<String, String>) val).get('type'),
+                                    ((Map<String, String>) val).get('description'))
+                            })
+
+                            String routeDocumentString = dumper.dumpToString(template)
+
+                            log.info('\n>>>   Converted Template as Yaml:\n{}', routeDocumentString)
+                            log.info('\nWriting project to: {}', thisKameletDirectory)
+
+                            Path propFile = Paths.get(thisKameletDirectory.toAbsolutePath().toString(),
+                                'props.json')
+                            // FIXME: Convert this to a project file defining the assembly
+                            Files.writeString(propFile, convertMapToJson(props))
+
```

Comment:
These calls & the methods that implement them are new... This is the bulk of the semantic change